src/shared/condition.c

```diff
@@ -239,6 +235,46 @@ static int condition_test_kernel_version(Condition *c, char **env) {
         return true;
 }
 
+static int condition_test_version(Condition *c, char **env) {
+        _cleanup_free_ char *word = NULL;
+        const char *p = c->parameter;
+        char ver[8], *verp;
+        struct utsname u;
+        const char *s;
+        int r;
+
+        assert(c);
+        assert(c->type == CONDITION_VERSION || c->type == CONDITION_KERNEL_VERSION);
+
+        r = extract_first_word(&p, &word, " =!<>$", 0);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to parse compare predicate \"%s\": %m", p);
+
+        /* An empty condition is true, but one where the right operand is missing is invalid. */
+        if (r == 0) {
+                if (!c->parameter[0])
+                        return true;
+                return -EINVAL;
+        }
+
+        s = strstrip(word);
+        p = c->parameter + strlen(s);
+
+        if (streq(s, "systemd")) {
+                xsprintf(ver, "%d", PROJECT_VERSION);
```

Comment:
Please just use STRINGIFY(PROJECT_VERSION). No need for runtime xsprintf() roundtrip